lib/auth_page.dart

```diff
@@ -66,6 +83,56 @@ Stream<int> _signin(BuildContext context) async* {
   }
 }
 
+void _facebookSignout() {
+  try {
+    final _facebookLogin = FacebookLogin();
+    _facebookLogin.logOut();
+  } catch (error, trace) {
+    // reset the UI and display an alert
+  }
+}
+
+Stream<int> _facebookSignin(BuildContext context) async* {
+  try {
+    final _fireAuth = FirebaseAuth.instance;
+    final facebookLogin = FacebookLogin();
+    final result = await facebookLogin.logIn(['email']);
+
+    switch (result.status) {
+      case FacebookLoginStatus.loggedIn:
+        /// the auth info will be picked up by the listener on [onAuthStateChanged]
+        /// and emitted by [streamOfStateChanges]
+        
+        // signal to change UI
+        yield 2;
+
+        final credential = FacebookAuthProvider.getCredential(accessToken: result.accessToken.token);
+        await _fireAuth.signInWithCredential(credential);
+
+        // we are signed in so reset the UI
+        yield 0;
+        break;
+      case FacebookLoginStatus.cancelledByUser:
+        // _showCancelledMessage();
+        yield 0;
+        break;
+      case FacebookLoginStatus.error:
```

Comment:
How about throwing here so the exception gets caught and displayed?

Reply:
As above. I followed the google example. It didn't have any exception handling why would this?

Reply:
If you throw from here it will get picked up by the catch block and if the catch block is set to pass the error message to the _showDialog function (as discussed in the previous comment) then that's all we would need to get the error message displayed.

Reply:
The reason I suggest we throw here is that this section is wrapped in a try catch and the catch is where are already dealing with errors so it just means less work.  There is a commented out call to a function here that I assume would be doing the same thing.

Reply:
Just one final point just because I think it's important - example code will often be missing things like exception handling, it's exactly the kind of thing that gets taken out of example code so the example can be shorter and just focus on what is specific to the thing being demonstrated.  I would definitely argue that if example code is missing exception handling that doesn't indicate exception handling is not needed.